packages/discord.js/src/managers/GuildMemberManager.js

```diff
@@ -254,6 +254,25 @@ class GuildMemberManager extends CachedManager {
     return data.reduce((col, member) => col.set(member.user.id, this._add(member, cache)), new Collection());
   }
 
+  /**
+   * The data for editing a guild member belonging to the client user.
+   * @typedef {Object} GuildMemberEditMeData
+   * @property {?string} [nick] The nickname to set for the client user in a guild
+   * @property {string} [reason] The reason for editing the client user's data in a guild
+   */
+
+  /**
+   * Edits the client user in a guild.
+   * @param {GuildMemberEditMeData} data The data to edit the guild member belonging to the client user with
+   * @returns {Promise<GuildMember>}
+   */
+  async editMe({ nick, reason }) {
```

Comment:
If all properties are optional, shouldn't the object itself also be optional?

Reply:
I could change it to `({ nick, reason }) = {}`, but the `edit()` method below is identical to this one. It also helps people to actually pass something to this method, as calling `editMe()` with no arguments would likely indicate a developer is doing something wrong (the method will throw additionally). `editMe({})` works, however, and is a no-op. That's why `data` is a required parameter here yet the properties in the object are optional.